tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -85,19 +92,67 @@ class resource_type(ArmResourceManager.resource_type):
 
 @CosmosDB.filter_registry.register('firewall-rules')
 class CosmosDBFirewallRulesFilter(FirewallRulesFilter):
+    """Filters resources by the firewall rules
+
+    Rules can be specified as x.x.x.x-y.y.y.y or x.x.x.x or x.x.x.x/y.
+
+    CosmosDB rules include keywords ``AzureCloud`` and ``Portal`` to provide similar experience with
+    Azure portal.
+
+    With the exception of **equal** all modes reference total IP space and ignore
+    specific notation.
+
+    **include**: True if all IP space listed is included in firewall.
+
+    **any**: True if any overlap in IP space exists.
+
+    **only**: True if firewall IP space only includes IPs from provided space
+    (firewall is subset of provided space).
+
+    **equal**: the list of IP ranges or CIDR that firewall rules must match exactly.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: cosmosdb-with-firewall
+                resource: azure.cosmosdb
+                filters:
+                  - type: firewall-rules
+                    include:
+                      - '131.107.160.2-131.107.160.3'
+                      - 10.20.20.0/24
+                      - AzureCloud
+                      - Portal
```

Comment:
Shouldn't this be a bypass field? And then we should actually remove it from the IP set we use for normal IP filtering and treat it as a separate thing?  That is what the action does.